navigator/runtime-compose/src/main/java/com/freeletics/mad/navigator/compose/NavigationSetup.kt

```diff
@@ -38,16 +36,16 @@ public fun NavigationSetup(navigator: NavEventNavigator) {
     }
 
     val backDispatcher = LocalOnBackPressedDispatcherOwner.current!!.onBackPressedDispatcher
-    DisposableEffect(lifecycleOwner, backDispatcher, navigator) {
-        backDispatcher.addCallback(lifecycleOwner, navigator.onBackPressedCallback)
+    DisposableEffect(backDispatcher, navigator) {
+        backDispatcher.addCallback(navigator.onBackPressedCallback)
 
         onDispose {
             navigator.onBackPressedCallback.remove()
         }
```

Comment:
We already call `remove` in `onDispose` so there is no need to pass a lifecycle to `addCallback` which would achieve the same thing.